src/blockchains/bip122.ts

```diff
@@ -1,13 +1,14 @@
-import SHA256 from 'crypto-js/sha256'
-import RIPEMD160 from 'crypto-js/ripemd160'
-import enc from 'crypto-js/enc-hex'
+import RIPEMD160 from 'ripemd160'
 import { bytesToBase58 } from '../util'
+import { sha256 } from '../Digest'
 
-export const publicKeyToAddress = (publicKeyBuffer: string): string => {
-  const publicKeyHash = RIPEMD160(SHA256(enc.parse(publicKeyBuffer)))
-  const step1 = Buffer.from('00' + publicKeyHash.toString(enc), 'hex')
-  const step2 = SHA256(SHA256(enc.parse(step1.toString('hex'))))
-  const checksum = step2.toString(enc).substring(0, 8)
-  const step3 = step1.toString('hex') + checksum
-  return bytesToBase58(Uint8Array.from(Buffer.from(step3, 'hex')))
+export const publicKeyToAddress = (publicKey: string): string => {
+  const publicKeyBuffer = Uint8Array.from(Buffer.from(publicKey, 'hex'))
+  const hash = new RIPEMD160().update(Buffer.from(sha256(publicKeyBuffer))).digest()
+  const step1 = Buffer.concat([Buffer.from('00', 'hex'), hash])
```

Comment:
Buffer error from ripemd160, and [ripemd160](https://github.com/crypto-browserify/ripemd160/blob/master/index.js) also use buffer.

```
ERROR in ./node_modules/ripemd160/index.js 2:13-37
Module not found: Error: Can't resolve 'buffer' in '/home/runner/work/did-jwt/did-jwt/node_modules/ripemd160'
```

So use `crypto-js/ripemd160` package (or [ripemd160.js](https://github.com/brix/crypto-js/blob/develop/src/ripemd160.js)) only, or looking for other way.

What do you think?

Reply:
Hm, thoughts @mirceanis ? Could `crypto-js/ripemd160` cause issues somewhere bc it's js modules?

Reply:
> Hm, thoughts @mirceanis ? Could `crypto-js/ripemd160` cause issues somewhere bc it's js modules?

I'm not sure. Usually react-native is a pain, but I'll have to check.

Reply:
make and add new `ripemd160.ts`.

Reply:
@daoauth I've been trying to test your changes in react-native (that's where most issues arise with new dependencies), and there are still issues with regards to the use of `Buffer`.
I suppose that they can be worked around by some rn-nodeify magic, but I would like to keep that option as a last resort, since it complicates all setups.

@oed's suggestion to use `uint8arrays` instead of `Buffer` refers to the complete replacement of the nodejs-specific `Buffer` API.

That being said, I'm not yet comfortable taking on the maintenance burden of a RIPEMD160 implementation.
Ideally there should be a Uint8Array friendly RIPEMD160 implementation we could use.
I haven't yet had time to try the variant with `crypto-js/ripemd160`; but I see there were some build issues with it.

Reply:
@mirceanis How about `DataView`? Can I use it?

Reply:
wow, that was fast :)
This new commit seems to work ok in my tests.
I'm still reluctant about taking on the RIPEMD160 implementation, but I'm also not aware of trustworthy alternatives.
@oed any insights appreciated ;)

Reply:
`DataView` seems like a standard lib: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView